### PR TITLE
gNOI-3.6_Kill_a_Daemon_test.md

### PR DESCRIPTION
**Pull request by sachendras**
- Add a gNOI negative test to verify health of a daemon after it is killed.
- Once the daemon is brought back, health of the daemon is rechecked to confirm functionality.

This test will start with validating config daemon behavior but is open to add other processes as they are found necessary to be tested.

### CODE REVIEW

**Review by dplore — COMMENTED** [review]
THis seems to highly overlap with https://github.com/openconfig/featureprofiles/tree/main/feature/gnoi/system/tests/copying_debug_files_test

Should we just update with the negative case you are proposing here?

File: feature/gnoi/system/tests/kill_a_daemon_test/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+This is a negative test to kill some Software daemons and verify if the implementation is able to accurately stream the Health and all related artifacts for the subject daemon.
+  
+## Procedure
+
+* gNOI-3.6.1 : Kill configuration daemon.
+
+  Kill the process that manages device configuration using the **gNOI.KillProcessRequest_SIGNAL_ABRT** and restart flag set to False. This will terminate the process and will also dump a Core file, while maintaing the process in a down state. Verify if the leaf /components/component/healthz/state/status transitioned to **UNHEALTHY**.
+   1. If the software module has a status of **UNHEALTHY**, issue healthZ.Get() to collect more details on the event. Also, use HealthZ.Artifact() to collect artifacts like core dump, logs etc.
+   2. Initiate a **gNOI.KillProcessRequest_SIGNAL_HUP** operation to restart and recover the killed process.
```
Comment (dplore) [inline feature/gnoi/system/tests/kill_a_daemon_test/README.md:13]: I don't think SIGNAL_HUP will start a process which has been stopped.  It doesn't look like gNOI has a feature to start a stopped process.  Probably you have to just work with SIGNAL_ABRT and restart=true.  See gNOI-3.5 which already does this (for a different process/bgp)

File: feature/gnoi/system/tests/kill_a_daemon_test/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# gNOI-3.6: Kill a Daemon Test
```
Comment (dplore) [inline feature/gnoi/system/tests/kill_a_daemon_test/README.md:1]: To differentiate from gNOI-3.5 which also kills a daemon and uses healthz:

```suggestion
# gNOI-3.6: Configuration daemon kill process  and healthz
```

File: feature/gnoi/system/tests/kill_a_daemon_test/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      * Nokia: 
+   * NOS implementations will need to model their agent that handles device configuration as a [" component of the type SOFTWARE_MODULE"](https://github.com/openconfig/public/blob/master/release/models/platform/openconfig-platform-types.yang#L394) and represent it under the componenets/component tree
+
+
+## Config Parameter Coverage
+*   HealthZ.Get()
+*   HealthZ.Artifact()
+
+## Telemetry Parameter Coverage
+*   /components/component[**process that handles configuration of the DUT**]/healthz/state/status/
```
Comment (dplore) [inline feature/gnoi/system/tests/kill_a_daemon_test/README.md:32]: Add

## Protocol/RPC Parameter Coverage

*   gNOI
    *   System
        *   KillProcess
    *   Healthz
        *   Get
        *   Check
        *   Artifact

**Comment by sachendras** [comment]
> THis seems to highly overlap with https://github.com/openconfig/featureprofiles/tree/main/feature/gnoi/system/tests/copying_debug_files_test
> 
> Should we just update with the negative case you are proposing here?

ACK, https://github.com/openconfig/featureprofiles/pull/2271 for updates to "copying_debug_files_test". I will close this one.

File: feature/gnoi/system/tests/kill_a_daemon_test/README.md (hunk 2 above)
Comment (sachendras) [reply feature/gnoi/system/tests/kill_a_daemon_test/README.md:1]: Incorporated in [copying_debug_files_test](https://github.com/openconfig/featureprofiles/pull/2271) instead.

File: feature/gnoi/system/tests/kill_a_daemon_test/README.md (hunk 3 above)
Comment (sachendras) [reply feature/gnoi/system/tests/kill_a_daemon_test/README.md:32]: Incorporated in [copying_debug_files_test](https://github.com/openconfig/featureprofiles/pull/2271) instead

**Comment by sachendras** [comment]
Test Incorporated in [copying_debug_files_test](https://github.com/openconfig/featureprofiles/pull/2271) file instead.

**Pull request closed without merging**